### ISSUE TITLE
CPU utilization rule

### ISSUE THREAD

**Issue by vshulkin** [target/pod-reaper#95]
Is it possible to add a CPU utilization rule? I’m trying to terminate pods that have been idle for a period of time. In a shared cluster environment, some users allocate resources but fail to use and resist releasing them, which prevents others from accessing those resources.

**Comment by brianberzins** [comment]
I don't think this one would work... at least not in a way that I think would actually be desirable.

My thoughts kinda boil down to two general ideas:
- getting the CPU metrics on pods
- understanding that metrics as a value over time

Right now, this is designed to really just look at just the moment of inspection. To get pods metrics, it could look each time it's evaluated, but we'd likely need to keep track of something like "has this pod ben CPU spiked for 1 second, 10 seconds, an hour or a week". I think the complexity of adding in metrics + looking over a time interval would put this out of the scope of this little project.
